src/analysis/bounded_variation.lean

```diff
@@ -207,6 +207,56 @@ begin
   simp [u, edist_comm],
 end
 
+lemma lower_continuous_aux {ι : Type*} {F : ι → α → E} {p : filter ι}
+  {f : α → E} {s : set α} (hF : tendsto_uniformly_on F f p s) :
+  ∀ (v : ℝ≥0∞), v < evariation_on f s → (∀ᶠ (n : ι) in p, v < evariation_on (F n) s) :=
+begin
+  rintro v hv,
+  rw emetric.tendsto_uniformly_on_iff at hF,
+  obtain ⟨v',vv',v'var⟩ := exists_between hv,
+  obtain ⟨⟨n,⟨u,um,us⟩⟩,hlt⟩ := lt_supr_iff.mp v'var,
+  let ε := (v' - v)/(2*n),
+  have hvp : 0 < v' - v, by { simp only [vv', tsub_pos_iff_lt],},
+  have : v' - v ≠ ⊤ := ennreal.sub_ne_top (ne_of_lt (lt_of_lt_of_le v'var le_top)),
+  have : v' = v + (v' - v) := (add_tsub_cancel_of_le vv'.le).symm,
+  have hε : 0 < ε := ennreal.div_pos_iff.mpr
+    ⟨hvp.ne.symm, ennreal.mul_ne_top ennreal.two_ne_top (ennreal.nat_ne_top n)⟩,
+  refine (hF ε hε).mono (λ i hi, _),
+  suffices h : v' < evariation_on (F i) s + (v' - v),
+  { nth_rewrite 0 this at h,
+    rw ←ennreal.add_lt_add_iff_right ‹v'-v≠⊤›,
+    exact h, },
+  calc v'
+     < ∑ (j : ℕ) in finset.range n, edist (f (u j.succ)) (f (u j))  : hlt
+  ...≤ ∑ (j : ℕ) in finset.range n, (edist (F i (u j.succ)) (F i (u j)) + 2*ε) : by
+  begin
+    refine finset.sum_le_sum (λ j jn, (edist_triangle4 _ (F i (u j.succ)) (F i (u j)) _).trans _),
+    rw [two_mul, ←add_assoc _ ε ε, add_comm _ ε, ←add_assoc],
+    refine add_le_add (add_le_add_right (hi (u j.succ) (us j.succ)).le _) _,
+    rw edist_comm, exact (hi (u j) (us j)).le,
+  end
+  ...= ∑ (j : ℕ) in finset.range n, edist (F i (u j.succ)) (F i (u j)) + 2*n*ε :
+  begin
+    simp only [finset.sum_add_distrib, finset.sum_const, finset.card_range, nsmul_eq_mul,
+               ←mul_assoc, mul_comm _ (2:ℝ≥0∞)],
+  end
+  ...≤ ∑ (j : ℕ) in finset.range n, edist (F i (u j.succ)) (F i (u j)) + (v' - v) :
+  begin
+    refine add_le_add_left (ennreal.mul_div_le) _,
+  end
+  ...≤ evariation_on (F i) s  + (v' - v) : add_le_add_right (sum_le (F i) _ um us) (v' - v)
+end
+
+lemma lower_continuous {s : set α} :
```

Comment:
```suggestion
lemma lower_continuous (s : set α) :
```
There is no way `s` can be inferred from another argument (as there is no other argument), so it's best to have it explicit. The exception to this rule is when you are proving an equivalence, and `s` can be inferred from both sides of the implication.

Reply:
If proving an equality, and `s` appears on both sides, can I also leave it implicit then? Asking since it seems it would be similar to the equivalence, and say end up the same when used e.g. in rewrites.